modules/phase_field/include/kernels/MultiGrainRigidBodyMotion.h

```diff
@@ -0,0 +1,30 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef MULTIGRAINRIGIDBODYMOTION_H
+#define MULTIGRAINRIGIDBODYMOTION_H
+
+#include "GrainRigidBodyMotionBase.h"
+
+//Forward Declarations
+class MultiGrainRigidBodyMotion;
+
+template<>
+InputParameters validParams<MultiGrainRigidBodyMotion>();
+
+class MultiGrainRigidBodyMotion : public GrainRigidBodyMotionBase
+{
+public:
+  MultiGrainRigidBodyMotion(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual();
+  virtual Real computeQpJacobian();
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar);
+  virtual Real computeQpCJacobian();
```

Comment:
Yes, ```computeQpJacobian``` could defer to ```computeQpJacobian```, but I think factoring out the calculation of the Jacobian entry of ```c``` is somewhat more readable code IMO.